tlslite/tlsconnection.py

```diff
@@ -424,6 +425,9 @@ def _handshakeClientAsyncHelper(self, srpParams, certParams, anonParams,
                 raise ValueError("Caller passed no nextProtos")
         if alpn is not None and not alpn:
             raise ValueError("Caller passed empty alpn list")
+        if serverName and not is_valid_hostname(serverName):
```

Comment:
No, the check is only for validity of server name. An empty name is completely valid, then the Server Name Indication extension is simply not sent (since an empty SNI hostname is an invalid encoding of the extension there's no point in differentiating between empty string and a None)